lib/chef/dsl/recipe.rb

```diff
@@ -105,27 +120,10 @@ def method_missing(method_symbol, *args, &block)
         end
       end
 
+      # This module is deprecated and will be removed in Chef 13
```

Comment:
If you make this `@deprecated Use Chef::DSL::Recipe instead, will be removed in Chef 13` it will show a nice banner in the YARD docs.